packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -173,6 +175,40 @@ File getUniqueFile(Directory dir, String baseName, String ext) {
   return _getUniqueFile(dir, baseName, ext);
 }
 
+/// Return [directory]'s total size in a human-readable format.
+Future<String?> getDirectorySize(Directory directory) async {
+  const LocalPlatform platform = LocalPlatform();
```

Comment:
the `Platform` and a `ProcessManager` should be injected into this function, so that they can be faked in tests.

Reply:
Another option might be to expose `du` in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/os.dart through the OS abstraction. Code like this already seems to live there.